### PR TITLE
feat(responsive): rem-based --breakpoint-* tokens

### LINKED ISSUE #99: [6/10] feat(responsive): rem-based --breakpoint-* tokens
## Summary

Ship 5 rem-based breakpoint CSS variables in the Tailwind `@theme` block (`--breakpoint-sm` … `--breakpoint-2xl`), plus a `--breakpoint-*: initial` reset, so Nexus **owns and declares** its breakpoint contract explicitly instead of silently inheriting Tailwind's.

**⚠ Correction (re-verified against Tailwind v4.1.18, May 2026):** the original framing ("Tailwind defaults to px; overriding to rem is the accessibility upgrade") was a **Tailwind v3 fact**. **Tailwind v4 already ships these exact rem breakpoints** (`--breakpoint-sm: 40rem` … `--breakpoint-2xl: 96rem`, see `node_modules/tailwindcss/theme.css`). So this is **not** a px→rem behavior change — at runtime it is net-zero at the default font size. The real wins are:

1. **Governance / consistency** — every other Nexus namespace (`--color-*`, `--spacing-*`, `--radius-*`, `--shadow-*`) is `initial`-reset and re-declared so Nexus owns the values. Breakpoints are currently the lone namespace inherited straight from Tailwind. This brings them under the same reset-and-own discipline and insulates Nexus if Tailwind ever changes its defaults.
2. **Explicit + documented + greppable** — the values live in `nexus.css` with rationale, not as implicit framework defaults a reader has to know about.
3. **Override surface** — a documented build-time override path for consumers (their own `@theme` block).
4. **`:root` exposure** — explicit declaration guarantees the `--breakpoint-*` variables are emitted as real CSS custom properties for non-Tailwind contexts (Tailwind uses its internal defaults to generate media queries but does not necessarily surface them as `:root` vars).

**⚠ Build-time, not runtime (post-spike, May 2026):** this enables **build-time** override, NOT runtime override. Setting `:root { --breakpoint-lg: 70rem }` at runtime does NOT change Tailwind's generated `nx:lg:` media queries — `@media` doesn't accept `var()` in any current browser, so Tailwind bakes the resolved value at build time. The same constraint applies to container queries. Consumers override by adding their own `@theme` block before/after the Nexus import, not by setting CSS variables at runtime.

## Research basis

- Phase 1 audit: [`reports/nexus-ds-intelligence.html § Tokens > Breakpoints`](../reports/nexus-ds-intelligence.html#sec-tokens-breakpoints) — Nexus ships zero **explicit** breakpoint tokens (it inherits Tailwind's).
- Phase 2 §5 Decision A2: [`reports/phase-2-token-decisions.html § 5`](../reports/phase-2-token-decisions.html#bp-decisions)
- Atlassian DS (`@atlaskit/primitives`), Shopify Polaris (`--p-breakpoints-*`), Mantine — all ship explicit rem/em-based breakpoint tokens rather than leaning on framework defaults. Declaring them explicitly (and owning the reset) is the alignment move.

## Scope

Add to the `@theme` block (via the token source + generators — `nexus.css` is auto-generated, do **not** hand-edit it):

```css
--breakpoint-sm: 40rem; /* 640px @ 16px root  · narrow */
--breakpoint-md: 48rem; /* 768px @ 16px root  · narrow */
--breakpoint-lg: 64rem; /* 1024px @ 16px root · Standard floor ★ */
--breakpoint-xl: 80rem; /* 1280px @ 16px root · Standard reference ★ */
--breakpoint-2xl: 96rem; /* 1536px @ 16px root · wide */
```

Values match Tailwind v4's own rem defaults exactly; declaring them explicitly brings them under Nexus's reset-and-own discipline. Tailwind v4 picks them up automatically (it scans `@theme` for `--breakpoint-*`).

**Implementation note:** mirror the z-index scale (#149) — a standalone semantic token file (`packages/core/tokens/semantic/breakpoints.json`) with direct values (no primitive layer, no modes, no light/dark), collected by a dedicated `collectBreakpointsTokens()` and emitted through the shared `generateThemeCSS()` into both `generate-tailwind-package.js` and `generate-modular.js`. Then regenerate `packages/tailwind/nexus.css` + playground `globals.css`.

## Acceptance criteria

- [ ] 5 `--breakpoint-*` CSS variables added to the `@theme` block, in rem, via the token source + generators (not by hand-editing `nexus.css`)
- [ ] `--breakpoint-*: initial` reset line added in the reset cluster at the top of the `@theme` block (matches existing convention at `nexus.css:21-24` for `--color-*: initial`, `--spacing-*: initial`, etc.)
- [ ] Tailwind generates correct `nx:sm:` / `nx:md:` / `nx:lg:` / `nx:xl:` / `nx:2xl:` utilities (verified via grep on generated CSS + the existing `nx:sm:` usages in `dialog.tsx` compiling under Storybook)
- [ ] Breakpoints behave correctly at default browser font-size (16px) — identical to today (values unchanged)
- [ ] Breakpoints behave correctly at increased browser font-size (e.g., 20px) — scale proportionally (inherent to rem)
- [ ] **Build-time override path documented**: a consumer adding their own `@theme { --breakpoint-lg: 70rem }` regenerates Tailwind utilities with the new value
- [ ] **Do NOT claim runtime override works** in documentation — `@media` doesn't accept `var()`; values are baked at build time
- [ ] No Storybook regression (zero react component changes; the reset re-adds identical values)
- [ ] Companion: display class documentation in [8/10]

## References

- Phase 2 artifact: [`reports/phase-2-token-decisions.html § 5 Decision A2`](../reports/phase-2-token-decisions.html#bp-decisions)
- Tailwind v4 `--breakpoint-*` docs: https://tailwindcss.com/docs/responsive-design
- Tailwind v4 default theme (already rem): `node_modules/tailwindcss/theme.css`
- Precedent: z-index scale #149 (standalone semantic token → `generateThemeCSS` → both generators)

Milestone: [Interaction & layering tokens · Phase X](https://github.com/nexuslabs-ai/nexus/milestone/4)

### PR DESCRIPTION
**Pull request by aishvaryagupta**
## Summary

- Ship 5 rem-based `--breakpoint-*` tokens (`sm`/`md`/`lg`/`xl`/`2xl` = `40`/`48`/`64`/`80`/`96rem`) plus a `--breakpoint-*: initial` reset into the Tailwind `@theme` block — **via the token source + generators**, mirroring the z-index scale (#149), not by hand-editing the generated `nexus.css`.
- New `packages/core/tokens/semantic/breakpoints.json` (standalone semantic — no primitive layer, no modes, no light/dark), collected by `collectBreakpointsTokens()` in `utils.js` and emitted through the shared `generateThemeCSS()` into both `generate-tailwind-package.js` and `generate-modular.js`.
- Regenerated `packages/tailwind/nexus.css` + playground `globals.css` (breakpoint-only diff; no brand/color/`package.json` drift).

### Rationale — corrected from the issue's original premise

The issue originally framed this as a **px → rem accessibility upgrade**. That's a **Tailwind v3 fact** — **Tailwind v4 already ships these exact rem breakpoints** (`node_modules/tailwindcss/theme.css`). I re-verified against v4.1.18 and **updated the #99 body** to fix the premise. So this is **net-zero at runtime**; the real value is:

1. **Governance / consistency** — every other namespace (`--color-*`, `--spacing-*`, `--radius-*`, `--shadow-*`) is `initial`-reset + re-declared so Nexus _owns_ the values. Breakpoints were the lone namespace inherited implicitly; this brings them under the same reset-and-own discipline and insulates Nexus if Tailwind changes its defaults.
2. **Explicit + documented + greppable** in `nexus.css`.
3. **Override surface** — a documented build-time path for consumers.
4. **`:root` exposure** — explicit declaration guarantees the vars are emitted for non-Tailwind contexts.

### Build-time, not runtime

Override is **build-time only** (consumer adds their own `@theme` block). `@media` can't read `var()`, so Tailwind bakes the resolved value at build time — documented in the token file's `$description`, and deliberately **not** claimed as runtime override.

### Out of scope

Display-class documentation + responsive demo page is deferred to **[8/10]** per the issue.

## GitHub Issue

Closes #99

## Test Plan

- [x] `yarn typecheck` — passes
- [x] Lint — real-source (`eslint packages apps scripts`) passes (repo-wide `yarn lint` only flags a local untracked `.claude/worktrees/` copy; clean checkout / CI is unaffected)
- [x] 5 new tests pass — `collectBreakpointsTokens` (collects rem dimensions, throws naming the token on non-dimension `$type` and non-rem unit) + `@theme` / playground `globals.css` emission assertions
- [x] **Compile-proof:** `nx:sm:`…`nx:2xl:` compile against the regenerated `nexus.css` → `@media (width >= 40/48/64/80/96rem)` (Tailwind v4 range syntax)
- [x] Regen diff is breakpoint-only — no brand/color/`package.json` drift
- [x] No React component changes → no Storybook regression (re-added values are identical to Tailwind v4 defaults; the existing `nx:sm:` usages in `dialog.tsx` resolve unchanged)

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

**Review by examlly-tech — COMMENTED** [review]
## 🤖 Principal Architect Review — APPROVED

Standalone semantic file with direct rem values correctly mirrors the z-index precedent (#149) — no primitive layer is the right call for a token with no mode/theme variance. Structured DTCG `{value, unit}` matches the on-disk convention for every other dimension token (size/radius primitives) and `formatTokenValue` already handles it. The `--breakpoint-*: initial` reset-and-redeclare is the documented Tailwind v4 pattern, placed correctly in the reset cluster; values match v4.1.18 defaults. Collector placement and symmetric wiring into both generators are consistent with precedent.

### Architectural Concerns

**Forward-looking — rem-dimension collector factoring before [7/10]** (`packages/core/scripts/utils.js:776`) — `collectBreakpointsTokens` is the first standalone rem-dimension-literal collector; container queries [7/10] will likely add a near-identical `--container-*` one. Decide when [7/10] lands whether a shared `collectRemDimensionTokens(file, prefix)` replaces the per-namespace copies — extracting for a single instance now would be premature.

### Links

- **PR:** https://github.com/nexuslabs-ai/nexus/pull/153
- **GitHub Issue:** #99

---

_Review performed against `.claude/rules/` guidelines_

File: packages/core/tokens/semantic/breakpoints.json [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+    "$type": "dimension",
+    "$description": "640px @ 16px root — narrow."
+  },
+  "breakpoint-md": {
+    "$value": {
+      "value": 48,
+      "unit": "rem"
+    },
+    "$type": "dimension",
+    "$description": "768px @ 16px root — narrow."
```
Comment (examlly-tech) [inline packages/core/tokens/semantic/breakpoints.json]: **Minor — duplicate label:** `breakpoint-md` (768px) is labeled "narrow", the same word used for `breakpoint-sm` (640px) above. Give `md` a distinct label so the scale's metadata reads as five steps, not four.

File: packages/core/scripts/__tests__/utils.test.js [hunk 2]
```diff
@@ -454,4 +455,61 @@ describe('utils', () => {
...
+          'breakpoint-lg': {
+            $value: { value: 1024, unit: 'px' },
+            $type: 'dimension',
+          },
+        },
+        (dir) => {
+          expect(() => collectBreakpointsTokens(dir)).toThrow(/breakpoint-lg/);
+        }
+      );
+    });
```
Comment (examlly-tech) [inline packages/core/scripts/__tests__/utils.test.js:513]: **Minor — missing edge case:** the malformed-dimension guard (`typeof dim !== 'object'`) also catches a missing `$value`, but no test exercises it. The z-index sibling covers this path (`utils.test.js:452`). Add a `{ $type: 'dimension' }`-only fixture asserting `.toThrow(/breakpoint-lg/)`.

**Review by examlly-tech — COMMENTED** [review]
## 🤖 SDE2 Review — MINOR CHANGES

`collectBreakpointsTokens` guard-clause style matches the established collector idiom; both throw paths name the offending token like `collectZIndexTokens`, and the `formatTokenValue(dim, 'dimension')` reuse is correct (rem/numeric validation runs before the call). Generator wiring and the `--breakpoint-*: initial` reset are symmetric across both generators. Two minor items below — both in scope for this PR per `no-follow-up-deferral.md`.

### Issues

| #   | Severity | File                                                | Issue                                                                                                                            | Suggestion                                                                    |
| --- | -------- | --------------------------------------------------- | -------------------------------------------------------------------------------------------------------------------------------- | ----------------------------------------------------------------------------- |
| 1   | ⚠️ Minor | `packages/core/tokens/semantic/breakpoints.json:17` | `breakpoint-md` (768px) reuses the "narrow" label already on `breakpoint-sm` (640px), so the scale reads as four steps, not five | Give `md` a distinct label                                                    |
| 2   | ⚠️ Minor | `packages/core/scripts/__tests__/utils.test.js:513` | No test for the missing-`$value` throw path; the z-index sibling covers it (`utils.test.js:452`)                                 | Add a `$type: 'dimension'`-only fixture asserting `.toThrow(/breakpoint-lg/)` |

### Links

- **PR:** https://github.com/nexuslabs-ai/nexus/pull/153
- **GitHub Issue:** #99

---

_Review performed against `.claude/rules/` guidelines_

**Review by examlly-tech — APPROVED** [review]
## 🤖 Principal Architect Follow-up Review — APPROVED

Nothing structural changed since the prior APPROVED review — the 10-line delta is a `$description` wording tweak plus a vitest fixture, both in SDE2's lane. Re-affirming APPROVED.

### Previous Issues

| # | Original Issue | Status |
| - | -------------- | ------ |
| 1 | Forward-looking — rem-dimension collector factoring before [7/10] (`packages/core/scripts/utils.js:776`) | Deferred (forward-looking, not an action item for this PR) |

### Links

- **PR:** https://github.com/nexuslabs-ai/nexus/pull/153
- **GitHub Issue:** #99

---

_Review performed against `.claude/rules/` guidelines_

**Review by examlly-tech — APPROVED** [review]
## 🤖 SDE2 Follow-up Review — APPROVED

### Unresolved Issues

| # | Original Issue | File | Status | Notes |
| --- | --- | --- | --- | --- |
| 1 | duplicate "narrow" label on breakpoint-md | `packages/core/tokens/semantic/breakpoints.json:17` | ✅ Fixed | "compact" distinct; 5-step series reads coherently |
| 2 | missing-$value throw test | `packages/core/scripts/__tests__/utils.test.js:515` | ✅ Fixed | Mirrors z-index sibling; exercises typeof-guard, regex matches |

All previous issues resolved.

### Links

- **PR:** https://github.com/nexuslabs-ai/nexus/pull/153
- **GitHub Issue:** #99

---

_Review performed against `.claude/rules/` guidelines_
